modules/interacting-serverless-apps.adoc

```diff
@@ -24,3 +24,56 @@ helloworld-go   http://helloworld-go.default.example.com   helloworld-go-4wsd2
 $ curl http://helloworld-go.default.example.com
 Hello World: Go Sample v1!
 ----
+
+= Interacting with a serverless application using HTTP2 / gRPC
+
+{product-title} routes do not support HTTP2, and therefore do not support gRPC as this is transported by HTTP2.
+If you use these protocols in your application, you must call the application using the ingressgateway directly.
+To do this you will need to find the ingressgateway's public address, and the application's specific host.
+
+.Procedure
+
+. Find the application host. See the instructions in _Interacting with your serverless application_.
+. The ingressgateway's public address can be determined using this command:
++
+----
+oc -n istio-system get svc istio-ingressgateway
+----
++
+The output will be similar to this example:
++
+----
+NAME                   TYPE           CLUSTER-IP      EXTERNAL-IP                                                             PORT(S)                                                                                                                                      AGE
+istio-ingressgateway   LoadBalancer   172.30.51.103   a83e86291bcdd11e993af02b7a65e514-33544245.us-east-1.elb.amazonaws.com   80:31380/TCP,443:31390/TCP,31400:31400/TCP,15029:30672/TCP,15030:30970/TCP,15031:32657/TCP,15032:32400/TCP,15443:30167/TCP,15020:32285/TCP   67m
+----
++
+The public address is surfaced in the `EXTERNAL-IP` field, and in this case would be:
++
+----
+a83e86291bcdd11e993af02b7a65e514-33544245.us-east-1.elb.amazonaws.com
+----
+. Manually set the host header of your HTTP request to the application’s host, but direct the request itself against the public address of the ingressgateway.
++
+Here is an example, using the information obtained from the previous steps.
++
+----
+$ curl -H "Host: helloworld-go.default.example.com" a83e86291bcdd11e993af02b7a65e514-33544245.us-east-1.elb.amazonaws.com
+Hello Go Sample v1!
+----
++
+You can also make a gRPC request by setting the authority to the application’s host, while directing the request against the ingressgateway directly.
++
+Here is an example of what that looks like in the Golang gRPC client.
++
+----
+grpc.Dial(
+    host,
+    grpc.WithAuthority(domain),
+    grpc.WithInsecure(),
+)
+----
+
+[NOTE]
+====
+Ensure that you append the respective port to both “host” and “domain” (80 by default).
```

Comment:
Maybe make clear in the note that this only applies to gRPC.

Reply:
I moved the note up so it makes more sense (above the example). Would it make sense just to show the example with these appended though to show the format?

Reply:
That indeed makes a lot of sense. The example should then look like this:

```golang
grpc.Dial(
    "a83e86291bcdd11e993af02b7a65e514-33544245.us-east-1.elb.amazonaws.com:80",
    grpc.WithAuthority("helloworld-go.default.example.com:80"),
    grpc.WithInsecure(),
)
```

Reply:
Perfect, thanks Markus! :slightly_smiling_face: